File: conversions/decimal-to-binary/toBinary.js
/**
 * Decimal to binary conversion (recursive)
 * @param {Number} dec base-10 number
 * @returns {Array} binary number as integer array
 */
function toBinary(dec) {
  let bin = [];
  let quotient = Math.floor(dec / 2.0);
  let remainder = dec % 2;
  if (quotient > 0) {
    bin = [...bin, ...toBinary(quotient)];
  }
  bin.push(remainder);
  return bin;
}

/**
 * Helper function to log binary array as string
 * @param {Array} seq
 * @returns {String}
 */
function asString(seq) {
  let str = "";
  seq.forEach((x) => {
    str += String(x);
  });
  return str;
}

/* Test */
[...new Array(16).keys()].map((x) => console.log(`decimal: ${x} --> binario ${asString(toBinary(x))}`));

File: conversions/decimal-to-binary/python/to_binary.py
def to_binary(dec):
    """ recursively convert decimal number to binary """
    bin = []
    quotient = dec // 2
    remainder = dec % 2
    if quotient > 0:
        bin += to_binary(quotient)
    bin.append(remainder)
    return bin


def as_str(seq):
    y = ''
    for x in seq:
        y += str(x)
    return y


if __name__ == '__main__':
    [print(as_str(to_binary(x))) for x in list(range(16))]
